compression/tools/test_slow_drain.py

```
@@ -0,0 +1,425 @@
+    access_log off;
+    default_type application/octet-stream;
+    server {{
+        listen 127.0.0.1:{args.port} sndbuf=16384;
+{locs}    }}
+}}
+""", encoding="utf-8")
+
+        # nginx's own stdout/stderr to a file, not a PIPE: nothing
+        # drains a pipe here, and on failure the tail is the diagnosis.
+        nlog_path = root / "logs" / "nginx-stdout.log"
+        nlog = open(nlog_path, "w", encoding="utf-8")
```

Ruff (0.16.2)

[warning] Use a context manager for opening files

(SIM115)